### PR TITLE
FEAT: implement unyt.unyt_quantity.from_string

### PR DESCRIPTION
**Pull request by neutrinoceros**
This adds a from_string method to the unyt_quantity class. I originally wrote it in a separate project where I need to parse quantities from configuration (text) files, then realized it would be useful to have it as part of the library.

I consider this a draft for now. The implementation works as intended in every case I could think of (valid as well as invalid ones), but I would like to add doc strings (w/ doctests) to the actual function.

### CODE REVIEW

**Comment by JBorrow** [comment]
This is extremely useful as an addition, thanks!

**Comment by neutrinoceros** [comment]
Linter tests are failing, I assume that #189 would fix this.

**Comment by matthewfeickert** [comment]
> Linter tests are failing, I assume that #189 would fix this.

Yeah, I just checked this branch out in my fork and rebased it on top of PR #190 and it is failing the flake8 and black linter tests.

```
unyt/tests/test_linters.py FF  
```

So once PR #190 is merged a rebase here with application of `black` and some small lke8 stuff should have everything green and shiny. :)

**Comment by neutrinoceros** [comment]
excellent, let's focus on #190 then !

**Comment by neutrinoceros** [comment]
rebased and ready for review !

File: unyt/array.py [hunk 1]
```diff
@@ -1325,6 +1337,50 @@ def to_pint(self, unit_registry=None):
...
+        if not re.match(_QUAN_REGEXP, v):
+            raise ValueError("Received invalid quantity expression '{}'.".format(s))
+        res = re.search(_NUMB_REGEXP, v)
+        num = res.group()
+        res = re.search(_UNIT_REGEXP, v[res.span()[1] :])
+        unit = res.group()
+        return float(num) * Unit(unit, registry=unit_registry)
+
+    def to_string(self):
+        # this is implemented purely for symetry's sake
```
Comment (jzuhone) [inline unyt/array.py:1381]: Typo in "symmetry"

Comment (neutrinoceros) [reply unyt/array.py]: fixed !

**Comment by jzuhone** [comment]
@matthewturk care to look at this?

**Comment by matthewturk** [comment]
@jzuhone @neutrinoceros this looks good to me, but I am not sure I am authoritative on the subject!  I think it's a nice addition.

**Comment by neutrinoceros** [comment]
Sorry I forced-pushed again. I only fixed the typo that John found the other day, and which I _thought_ I fixed already.

**Comment by jzuhone** [comment]
Hi @neutrinoceros -- I almost forgot--can you add a short paragraph about this to the docs and an example or two?

**Comment by neutrinoceros** [comment]
Sure thing

**Pull request merged**